lib/portlayer/exec/base.go

```diff
@@ -461,12 +461,26 @@ func (c *containerBase) waitForSession(ctx context.Context, id string) error {
 	return c.waitFor(ctx, key)
 }
 
+// WaitForExec will wait for any kind of change in the state of a task. Then it will return signifying that the a refreshed handle must be returned.
+// users can then inspect that handle to examine and act against the changes in the task.
 func (c *containerBase) waitForExec(op trace.Operation, id string) error {
 	defer trace.End(trace.Begin(id, op))
 
-	// guestinfo key that we want to wait for
-	key := extraconfig.CalculateKeys(c.ExecConfig, fmt.Sprintf("Execs.%s.Started", id), "")[0]
-	return c.waitFor(op, key)
+	// guestinfo keys that we want to wait for
+	startedKey := extraconfig.CalculateKeys(c.ExecConfig, fmt.Sprintf("Execs.%s.Started", id), "")[0]
+	startKey := extraconfig.CalculateKeys(c.ExecConfig, fmt.Sprintf("Execs.%s.Detail.StartTime", id), "")[0]
+	stopKey := extraconfig.CalculateKeys(c.ExecConfig, fmt.Sprintf("Execs.%s.Detail.StopTime", id), "")[0]
+
+	// targeted exec
+	execConf := c.ExecConfig.Execs[id]
+
+	// construct key filter for exec state changes
+	filter := make(map[string]string)
+	filter[startedKey] = execConf.Started
+	filter[startKey] = fmt.Sprintf("%d", execConf.Detail.StartTime)
+	filter[stopKey] = fmt.Sprintf("%d", execConf.Detail.StopTime)
+
+	return c.vm.WaitForKeyChange(op, filter)
```

Comment:
I think this function should be explicit about which condition it's waiting for (`started != ""`).

Reply:
not 100% what you mean by this comment. The wait looks for a change from the handle that it was provided. so if you provide a handled that has `starterd == ""` and a change has occurred in started then it will return. the user can either inspect the task or atleast understand that if you wait right after creating and it is reasonable to assume that the task has atleast started.